src/apigateway/commands/invokeRemoteRestApi.ts

```diff
@@ -166,11 +168,21 @@ export function createMessageReceivedFunc({
                 outputChannel.appendLine(response.body!!)
             } catch (e) {
                 const error = e as Error
+                result = 'Failed'
                 outputChannel.appendLine(`There was an error invoking`)
                 outputChannel.appendLine(error.toString())
                 outputChannel.appendLine('')
             } finally {
                 postMessage({ command: 'invokeApiFinished' })
+                let method: string | undefined = undefined
+                // only set method if it is not empty or undefined
+                if (message.selectedMethod) {
+                    method = message.selectedMethod.toUpperCase()
+                }
```

Comment:
could avoid some ceremony, and `let`, by using a ternary, without losing clarity

    const method = message.selectMethod ? message.selectedMethod.toUpperCase() : undefined

Reply:
Does const method = `message.selectMethod?.toUpperCase()` work?

Reply:
Changed

Reply:
`Does const method = message.selectMethod?.toUpperCase() work?` Nope because we want null or empty